framework/platform/bpl/include/bpl/bpl_db.h

```diff
@@ -0,0 +1,97 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _BPL_DB_H_
+#define _BPL_DB_H_
+
+#include "bpl.h"
+#include "bpl_err.h"
+#include <stdint.h>
+#include <string>
+#include <unordered_map>
+
+namespace beerocks {
+namespace bpl {
+
+/****************************************************************************/
+/******************************* Definitions ********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************* Structures *********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************** Functions *********************************/
+/****************************************************************************/
+
+/**
+ * @brief Add parameter to the presistent DB
+ * 
+ * @param path name of the requested section.
+ * @param object full name of the parameter.
+ * @param val Initial value of the parameter.
+ * @return true on success, false otherwise.
+ */
+bool db_add_param(const std::string &path, const std::string &object, const std::string &val);
+bool db_add_param(const std::string &path, const std::string &object, const int &val);
+/**
+ * @brief Set a parameter given a specific path.
+ * 
+ * @param path name of the requested section.
+ * @param object full name of the parameter.
+ * @param val [in] value of the given parameter.
```

Comment:
For const reference need to write the oxygen like this;
`@param[in] param_name` and not `@param param_name [in]`

Also, it is required to add the direction for all  **non-const** references, so here it is not required.
Anyway, if you decide to do it also to **const** reference arguments, it is probably best if you do it on all of them in the file.

This comment apply to all the changes you've done in this commit.

Reply:
updated the oxygen according to the rest of the BPL apis